src/object/srv_obj_migrate.c

```diff
@@ -647,6 +725,29 @@ mrone_recx_vos2_daos(struct migrate_one *mrone, int shard, daos_iod_t *iods, int
 	mrone_recx_daos_vos_internal(mrone, false, shard, iods, iods_num);
 }
 
+static int
+mrone_obj_fetch_internal(struct migrate_one *mrone, daos_handle_t oh, d_sg_list_t *sgls,
+			 daos_iod_t *iods, int iod_num, daos_epoch_t eph, uint32_t flags,
+			 d_iov_t *csum_iov_fetch, struct migrate_pool_tls *tls)
+{
+	int rc;
+
+retry:
+	rc = dsc_obj_fetch(oh, eph, &mrone->mo_dkey, iod_num, iods, sgls,
+			   NULL, flags, NULL, csum_iov_fetch);
+	if (rc == -DER_TIMEDOUT &&
+	    tls->mpt_version + 1 >= tls->mpt_pool->spc_map_version) {
```

Comment:
hmm, tls->mpt_version is rebuild version. and spc_map_version is pool map version.   This check means if pool_map version does not change (increased), then it will keeps retry for timeout, otherwise it will fail.

+1 is because the rank status change will also update the pool map version, but it does not need rebuild. see update_one_tgt/dom().

Reply:
right, seems impossible to be "tls->mpt_version + 1 > tls->mpt_pool->spc_map_version", maybe I am wrong.